### PR TITLE
Added locked error message

### PR DESCRIPTION
**Pull request by jacob-nv**
## Description

Added error message in failed attempt of uploading to locked folder

## Related Issue

- Fixes https://github.com/owncloud/web/issues/5741

## Screenshots:
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Review by JammingBen — CHANGES_REQUESTED** [review]
We already have some error handling in place here: https://github.com/owncloud/web/blob/master/packages/web-runtime/src/components/UploadInfo.vue#L630. Could you please add it there?

Also, please add a changelog item 🙂 

Edit: Pipeline seems to fail because of some unrelated change, rebasing with current master should fix it though.

**Comment by jacob-nv** [comment]
> We already have some error handling in place here: https://github.com/owncloud/web/blob/master/packages/web-runtime/src/components/UploadInfo.vue#L630. Could you please add it there?
> 
> Also, please add a changelog item 🙂
> 
> Edit: Pipeline seems to fail because of some unrelated change, rebasing with current master should fix it though.

Not able to move the error message. I'm getting null responseCode there.

**Comment by JammingBen** [comment]
I see. Can you maybe retrieve the error for the file via `this.errors[file.meta.uploadId]` and then check for status code `423`? I'd really like to have to error handling in one place here.

**Comment by kulmann** [comment]
> I see. Can you maybe retrieve the error for the file via `this.errors[file.meta.uploadId]` and then check for status code `423`? I'd really like to have to error handling in one place here.

Fully agree UX-wise. The error in the title section of the upload info widget needs to stay short.

**Comment by jacob-nv** [comment]
> > I see. Can you maybe retrieve the error for the file via `this.errors[file.meta.uploadId]` and then check for status code `423`? I'd really like to have to error handling in one place here.
> 
> Fully agree UX-wise. The error in the title section of the upload info widget needs to stay short.

What would convey the message properly?
```
Upload failed: Locked
Upload failed: Folder is locked
```

**Comment by jacob-nv** [comment]
> I see. Can you maybe retrieve the error for the file via `this.errors[file.meta.uploadId]` and then check for status code `423`? I'd really like to have to error handling in one place here.

But you still have 2 places where you check for errors. Title of the window and then individual items.
I added error message for that "window" title, you are asking about error message for individual items. So should it be applied to both or only the items while window title has generic "Upload failed"?

**Comment by jacob-nv** [comment]
Something like this? @JammingBen @kulmann 
image

File: packages/web-runtime/src/components/UploadInfo.vue [hunk 1]
```diff
@@ -226,9 +226,9 @@ export default defineComponent({
         return this.$gettext('Upload cancelled')
       }
       if (Object.keys(this.errors).length) {
-        const isLocked = Object.values(this.errors).some((error: any) => error.statusCode === 423)
+        const isLocked = Object.values(this.errors).some((error: any) => error?.statusCode === 423)
         if (isLocked) {
-          return this.$gettext('Upload failed because the current folder is locked.')
+          return this.$gettext('Upload failed: Locked')
```
Comment (JammingBen) [inline packages/web-runtime/src/components/UploadInfo.vue:231]: I'd say it's sufficient to have the detailed error message down below, no need to handle the lock information here as well.

File: packages/web-runtime/src/components/UploadInfo.vue [hunk 2]
```diff
@@ -626,6 +630,9 @@ export default defineComponent({
       }
 
       const errorObject = formatErrorMessageToObject(error.message)
+      if (this.errors[item.meta.uploadId]?.statusCode === 423) {
+        return this.$gettext('Upload folder is locked')
```
Comment (JammingBen) [inline packages/web-runtime/src/components/UploadInfo.vue:634]: @kulmann @tbsbdr Suggestions for text? How about `The folder you're uploading to is locked`?

File: packages/web-runtime/src/components/UploadInfo.vue (hunk 1 above)
Comment (jacob-nv) [reply packages/web-runtime/src/components/UploadInfo.vue:231]: Got it

File: packages/web-runtime/src/components/UploadInfo.vue (hunk 2 above)
Comment (kulmann) [reply packages/web-runtime/src/components/UploadInfo.vue:630]: yep, works for me 👍

**Review by JammingBen — APPROVED** [review]
Nice 👍 

Needs another rebase with current master though, there have been quite a few pipeline changes recently.

**Pull request merged**